### PR TITLE
feat: transfer dust to Treasury

### PR DESCRIPTION
**Pull request by Roznovjak**

### CODE REVIEW

File: integration-tests/src/dust.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	Hydra::execute_with(|| {
+		let hdx_ed = <hydradx_runtime::Runtime as pallet_balances::Config>::ExistentialDeposit::get();
+
+		assert!(!hdx_ed.is_zero());
+
+		let transfer_amount = hdx_ed.checked_sub(1).unwrap();
+
+		// set Treasury balance to ED so it's not dusted
+		assert_ok!(Balances::set_balance(
```
Comment (enthusiastmartin) [inline integration-tests/src/dust.rs:21]: dont set ED for treasury account.

the treasury account cannot be dusted. You can assert that as well, that it is possible to have < ED in the treasury account and it wont get dusted.

Comment (Roznovjak) [reply integration-tests/src/dust.rs:21]: Well, no. The treasury account is dusted if the account doesn't exist, and the total issuance is decreased by the amount.

Comment (Roznovjak) [reply integration-tests/src/dust.rs:21]: The balance pallet (and the treasury pallet) are trying to `deposit_creating` to the treasury account (which doesn't exist yet) with the amount that is below ED, it results in no-op, and the amount id `Drop`ped.

Comment (enthusiastmartin) [reply integration-tests/src/dust.rs:21]: i see , it is strange. it feels like it should be allowed.

Comment (Roznovjak) [reply integration-tests/src/dust.rs:21]: Yes. The problem is that there is not a whitelist. Just a handler. So they are trying to solve dusting by sending dust somewhere.

**Review by enthusiastmartin — APPROVED** [review]

**Review by mrq1911 — APPROVED** [review]

**Comment by enthusiastmartin** [comment]
bump version and merge

**Pull request merged**